File: Sorting/Easy/LET_no1200_Minimum Absolute Difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int diff = arr[1] - arr[0];
        vector<vector<int>> res = {{arr[0], arr[1]}};
        
        for (int i = 2; i < arr.size(); i++) {
            if (arr[i] - arr[i - 1] == diff)
                res.push_back({arr[i - 1], arr[i]});
            else if (arr[i] - arr[i - 1] < diff) {
                diff = arr[i] - arr[i - 1];
                res = {{arr[i - 1], arr[i]}};
            }
        }
        
        return res;
    }
};

File: Sorting/Easy/LET_no1200_Minimum Absolute Difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        res = [[arr[0], arr[1]]]
        diff = arr[1] - arr[0]
        
        for i in range(2, len(arr)):
            if arr[i] - arr[i - 1] == diff:
                res.append([arr[i - 1], arr[i]])
            elif arr[i] - arr[i - 1] < diff:
                diff = arr[i] - arr[i - 1]
                res = [[arr[i - 1], arr[i]]]
        
        return res